data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -1,286 +1,9 @@
-
-        final GetIndexTemplatesRequest request = new GetIndexTemplatesRequest(indexTemplateName);
-        final GetIndexTemplatesResponse response = restHighLevelClient.indices().getIndexTemplate(request, RequestOptions.DEFAULT);
-
-        if (response.getIndexTemplates().size() == 1) {
-            return Optional.of(response.getIndexTemplates().get(0));
-        } else {
-            throw new RuntimeException(String.format("Found multiple index templates (%s) result when querying for %s",
-                    response.getIndexTemplates().size(),
-                    indexTemplateName));
-        }
-    }
-
-    private boolean shouldCreateIndexTemplate(final String indexTemplateName) throws IOException {
-        final Optional<IndexTemplateMetadata> indexTemplateMetadataOptional = getIndexTemplateMetadata(indexTemplateName);
-        if (indexTemplateMetadataOptional.isPresent()) {
-            final Integer existingTemplateVersion = indexTemplateMetadataOptional.get().version();
-            LOG.info("Found version {} for existing index template {}", existingTemplateVersion, indexTemplateName);
-
-            final int newTemplateVersion = (int) openSearchSinkConfiguration.getIndexConfiguration().getIndexTemplate().getOrDefault("version", 0);
-
-            if (existingTemplateVersion != null && existingTemplateVersion >= newTemplateVersion) {
-                LOG.info("Index template {} should not be updated, current version {} >= existing version {}",
-                        indexTemplateName,
-                        existingTemplateVersion,
-                        newTemplateVersion);
-                return false;
-
-            } else {
-                LOG.info("Index template {} should be updated from version {} to version {}",
-                        indexTemplateName,
-                        existingTemplateVersion,
-                        newTemplateVersion);
-                return true;
-            }
-        } else {
-            LOG.info("Index template {} does not exist and should be created", indexTemplateName);
-            return true;
-        }
-    }
-
-    //To suppress warnings on casting index template settings to Map<String, Object>
-    @SuppressWarnings("unchecked")
-    private void attachPolicy(
-            final IndexConfiguration configuration, final String ismPolicyId, final String rolloverAlias) {
-        configuration.getIndexTemplate().putIfAbsent("settings", new HashMap<>());
-        if (ismPolicyId != null) {
-            ((Map<String, Object>) configuration.getIndexTemplate().get("settings"))
-                    .put(IndexConstants.ISM_POLICY_ID_SETTING, ismPolicyId);
-        }
-        ((Map<String, Object>) configuration.getIndexTemplate().get("settings"))
-                .put(IndexConstants.ISM_ROLLOVER_ALIAS_SETTING, rolloverAlias);
-    }
 
+public interface IndexManager{
+	void setupIndex() throws IOException;
+	String getIndexName(final String indexAlias) throws IOException;
```

Comment:
Why did you add two overloads for `getIndexName()`? This could be somewhat confusing because it is unclear which method a caller (ie. `OpenSearchSink`) should use.

Reply:
@dlvenable, I thought it is easier to use this method when it is known that the indexAlias is static one. Also, this is equivalent to default behavior of getIndexName() where the index alias is taken from the config. Do you prefer to not have this? Or make it Optional?

Reply:
The reason is that the `getIndexName(String)` is the only method exposed to external clients. So from an external perspective it is unclear which to call and what the results will be.

Reply:
Removed the getIndexName() API.